src/App.tsx

```diff
@@ -1,223 +1,252 @@
+      setActiveTodosIds(todosIds => [...todosIds, ...activeIds]);
+      await Promise.all(activeTodos
+        .map(todo => updateTodoCompleted(todo.id, true)));
+      setActiveTodosIds([]);
+
+      if (completedTodos.length === todos.length) {
+        setActiveTodosIds(todosIds => [...todosIds, ...allIds]);
+        await Promise.all(todos
+          .map(todo => updateTodoCompleted(todo.id, !todo.completed)));
+        setActiveTodosIds([]);
+      }
+    } catch {
+      setNotification(Notifications.Update);
+      setActiveTodosIds([]);
+
+      setTimeout(() => {
+        setNotification(Notifications.None);
+      }, 3000);
+    }
+  };
+
+  const filteredTodos = filterTodos(filter);
 
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
         <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
-
-        <section className="todoapp__main" data-cy="TodoList">
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                defaultChecked
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">HTML</span>
+          {!!filteredTodos.length && (
             <button
+              data-cy="ToggleAllButton"
               type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">CSS</span>
+              className={classNames(
+                'todoapp__toggle-all',
+                { active: completedTodos.length === todos.length },
+              )}
+              onClick={toggleAllTodos}
+            />
+          )}
+          <NewTodo
+            newTodoField={newTodoField}
+            title={title}
+            setTitle={setTitle}
```

Comment:
Check your prop handlers names. According to convention they should start with 'on', e.g. 'onSetTitle'.
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

Reply:
agree